vision/module/location.py

```diff
@@ -34,9 +34,37 @@ def __init__(self):
         self.lower_bound = np.array(0)  # lower bound of slopes
         self.num_buckets = np.array(0)  # number of buckets applied to slopes
 
+    ## Determining if Module is in frame
+
+    def isInFrame(self) -> bool:
+        """
+        Determines if the Module is in the frame
+
+        Returns
+        -------
+        bool - true if module is in the frame and false if module is not in the frame
+        """
+        MIN_SLOPES_IN_BUCKET = (
+            15  # Minimum number of slopes per bucket to identify the module
+        )
+        MAX_CIRCLES = 100  # maximum number of circles that are allowed to be detected before in_frame fails
+
+        # Circle Detection
+        self._circleDetection()
+        if (
+            self.circles is None or self.circles.shape[0] > MAX_CIRCLES
+        ):  # no circles found or too many circles found
+            return False
+
+        # Get slopes and group parallel slopes
+        self._getSlopes()
+        self._groupSlopes()
+
+        return any(self.slope_heights > MIN_SLOPES_IN_BUCKET)
+
     ## Finding Distance to Module
 
-    def getDistance(self):
+    def getDistance(self) -> int:
```

Comment:
The getDistance function can be removed from location. A more robust calculation is performed by get_module_depth.py